products/sle/main.pm

```diff
@@ -418,6 +419,47 @@ logcurrentenv(
       SLE_PRODUCT SPLITUSR VIDEOMODE)
 );
 
+sub parse_condition {
+    my ($schedule, $condition_name) = @_;
+    # Get condition hash
+    my $condition = $schedule->{conditional_schedule}->{$condition_name};
+    # Iterate over variables in the condition
+    foreach my $var (keys %{$condition}) {
+        next unless my $val = get_var($var);
```

Comment:
Since the schedule and the conditions are separate (I guess it's the lesser of two evils?)
I would have written this as:
```suggestion
if (my $val = get_var($var)){
    for my $test (@{$condition->{$var}->{$val}}) {
        loadtest $test;
    }
} else {
 diag "$var does not have modules to be loaded";
}
```